src/lib/eina/eina_inline_thread_win32.x

```diff
@@ -0,0 +1,168 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library;
+ * if not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef EINA_INLINE_THREAD_WIN32_H_
+#define EINA_INLINE_THREAD_WIN32_H_
+
+
+
+typedef struct _Eina_Win32_Thread_Func
+{
+  void *data;
+  void *(*func)(void *data);
+
+} Eina_Win32_Thread_Func;
+
+typedef struct _Eina_Win32_Thread_Attr
+{
+  LPSECURITY_ATTRIBUTES lpThreadAttributes;
+  SIZE_T dwStackSize;
+  DWORD dwCreationFlags;
+} Eina_Win32_Thread_Attr;
+
+static inline void *
+_eina_thread_join(Eina_Thread t)
+{
+
+  int ret = (int)WaitForSingleObject(t, INFINITE);
+
+  if (ret != 0) return ret;
+  return NULL;
+}
+
+DWORD WINAPI _eina_thread_func(void *params)
+{
+  return (DWORD)((Eina_Win32_Thread_Func *)params)->func((void *)((Eina_Win32_Thread_Func *)params)->data);
+}
+
+static inline void _eina_thread_set_priority(Eina_Thread_Priority prio, Eina_Thread *t)
+{
+
+  int nPriority = THREAD_PRIORITY_NORMAL;
+
+  switch (prio)
+    {
+      case EINA_THREAD_URGENT:
+        nPriority = THREAD_PRIORITY_HIGHEST;
+        break;
+      case EINA_THREAD_NORMAL:
+        nPriority = THREAD_PRIORITY_NORMAL;
+        break;
+      case EINA_THREAD_BACKGROUND:
+        nPriority = THREAD_PRIORITY_BELOW_NORMAL;
+        break;
+      case EINA_THREAD_IDLE:
+        nPriority = THREAD_PRIORITY_IDLE;
+        break;
+    }
+
+  SetThreadPriority((HANDLE)*t, nPriority);
+}
+
+static inline Eina_Bool
+_eina_thread_create(Eina_Thread *t, int affinity, void *(*func)(void *data), void *data)
+{
+  Eina_Bool ret;
+  Eina_Win32_Thread_Attr thread_attr;
+
+  SECURITY_ATTRIBUTES sec_attributes;
+
+  sec_attributes.nLength = sizeof(SECURITY_ATTRIBUTES);
+  sec_attributes.lpSecurityDescriptor = NULL;
+  sec_attributes.bInheritHandle = EINA_TRUE;
+
+  thread_attr.lpThreadAttributes = &sec_attributes;
+
+  thread_attr.dwStackSize = 4000;
+  thread_attr.dwCreationFlags = 0;
+
+  LPDWORD threadID;
+
+  Eina_Win32_Thread_Func thread_func;
+  Eina_Thread_Call *c = (Eina_Thread_Call *)(data);
+
+  thread_func.func = func;
+  thread_func.data = data;
+
+  *t = CreateThread(thread_attr.lpThreadAttributes, thread_attr.dwStackSize, &_eina_thread_func, &thread_func, thread_attr.dwCreationFlags, threadID);
+
+  _eina_thread_set_priority(c->prio, t);
+
+  ret = (*t != NULL) ? EINA_TRUE : EINA_FALSE;
+
+  if (affinity >= 0 && ret) SetThreadAffinityMask(*t, (DWORD_PTR *)&affinity);
+
+  return ret;
+}
+
+static inline Eina_Bool
+_eina_thread_equal(Eina_Thread t1, Eina_Thread t2)
+{
+  DWORD t1_thread_id = GetThreadId(t1);
+  DWORD t2_thread_id = GetThreadId(t2);
+
+  return (t1_thread_id == t2_thread_id) ? EINA_TRUE : EINA_FALSE;
+}
+
+static inline Eina_Thread
+_eina_thread_self(void)
+{
+  return (Eina_Thread)GetCurrentThread();
+}
```

Comment:
Eina_Thread will be HANDLE right? Do you still need this cast?

Reply:
Oh, right